src/new-components/heroes/blog-hero.tsx

```diff
@@ -0,0 +1,83 @@
+import { GatsbyImage, getImage, IGatsbyImageData } from 'gatsby-plugin-image';
+import React, { useEffect, useState } from 'react';
+import { HeroContainer, TextContainer } from './support';
+import { HeroText, HeroWithText } from './hero-text';
+import styled from 'styled-components';
+import { useMedia } from 'react-use';
+
+const coverContainerCss = {
+  gridArea: '1/1',
+  /**
+   * make sure the image, which is positioned absolute by gatsby,
+   * is behind any content on the current stacking context.
+   */
+  zIndex: -1,
+};
+
+type ImageHeroProps = Partial<HeroWithText> & {
+  image: IGatsbyImageData;
+  imageSquare: IGatsbyImageData;
+  attribution?: any;
+};
+
+const AttributionContainer = styled.div`
+  position: absolute;
+  bottom: 0;
+  right: 0;
+  background-color: #fff;
+  display: block;
+  padding: 2px;
+  font-size: 0.8em;
+`;
+
+const Attribution = ({ attribution }) => (
+  <AttributionContainer>
+    Photo by{' '}
+    <a rel="nofollow noreferrer" target="_blank" href={attribution.source}>
+      {attribution.creator}
+    </a>
+  </AttributionContainer>
+);
+
+/**
+ * Display any given gatsby image as a hero image.
+ */
+export const BlogHero = ({
+  image,
+  imageSquare,
+  attribution,
+  title,
+  children,
+  kicker,
+}: ImageHeroProps) => {
+  const gatsbyImageData = getImage(image);
+  const gatsbyImageDataSquare = getImage(imageSquare);
+
+  const isSquare = useMedia('(max-aspect-ratio: 3/4)');
+
+  return (
+    <HeroContainer>
+      {isSquare && gatsbyImageDataSquare && (
+        <GatsbyImage
+          style={coverContainerCss}
+          alt=""
+          image={gatsbyImageDataSquare}
+        />
+      )}
+
+      {!isSquare && gatsbyImageData && (
+        <GatsbyImage style={coverContainerCss} alt="" image={gatsbyImageData} />
+      )}
+
+      {title && (
+        <TextContainer dimmed>
+          <HeroText title={title} kicker={kicker}>
+            {children}
+          </HeroText>
+        </TextContainer>
+      )}
+
+      {attribution && <Attribution attribution={attribution} />}
+    </HeroContainer>
+  );
+};
```

Comment:
`Layout/Hero` because it's made for page building. I would like to group the thing we need to create pages (heroes, containers, sections or so). That's why I just try things like that. This will change for sure. I will provide the missing story, which I forgot! Thanks for the ctahc.

Reply:
https://deploy-preview-273--satellytes-website-storybook.netlify.app/?path=/docs/layout-heroes--page
👍
![Screen Shot 2021-12-15 at 16 34 46](https://user-images.githubusercontent.com/1701755/146216241-38582b99-161b-452c-8ed4-27b8a73d043c.png)